leakcanary-android-leak-deobfuscation-plugin/src/main/java/com/squareup/leakcanary/leak/deobfuscation/plugin/CopyObfuscationMappingFileTask.kt

```diff
@@ -0,0 +1,53 @@
+package com.squareup.leakcanary.leak.deobfuscation.plugin
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyObfuscationMappingFileTask : DefaultTask() {
+
+  @Input
+  var variantDirName: String? = null
+
+  @Input
+  @Optional
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mappingFile: File? = null
+
+  @Input
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mergeAssetsDirectory: File? = null
+
+  @get:OutputFile
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  val leakCanaryAssetsOutputFile: File
+    get() = File(mergeAssetsDirectory, "leakCanaryObfuscationMapping.txt")
+
+  init {
+    description = "Puts obfuscation mapping file in assets directory."
+  }
+
+  @TaskAction
+  fun copyObfuscationMappingFile() {
+    mappingFile?.let { mappingFile ->
+      if (mappingFile.exists()) {
```

Comment:
can you skip the task if the mapping file or mergeAssetsDirectory is empty ?
https://mrhaki.blogspot.com/2017/02/gradle-goodness-skip-task-when-input.html

Reply:
I did it but `@SkipWhenEmpty` handles only empty files, passing `null` results in an exception so I had to add an empty file for a case where `variant.mappingFile` is null.